units/meson.build

```diff
@@ -3,322 +3,673 @@
-         'sockets.target.wants/'],
-        ['systemd-journal-catalog-update.service', '',
-         'sysinit.target.wants/'],
-        ['systemd-journal-flush.service',       '',
-         'sysinit.target.wants/'],
-        ['systemd-journal-gatewayd.socket',     'ENABLE_REMOTE HAVE_MICROHTTPD'],
-        ['systemd-journal-remote.socket',       'ENABLE_REMOTE HAVE_MICROHTTPD'],
-        ['systemd-journald-audit.socket',       ''],
-        ['systemd-journald-dev-log.socket',     '',
-         'sockets.target.wants/'],
-        ['systemd-journald.socket',             '',
-         'sockets.target.wants/'],
-        ['systemd-kexec.service',               ''],
-        ['systemd-machine-id-commit.service',   '',
-         'sysinit.target.wants/'],
-        ['systemd-journald@.socket',            ''],
-        ['systemd-journald-varlink@.socket',    ''],
-        ['systemd-networkd.socket',             'ENABLE_NETWORKD'],
-        ['systemd-poweroff.service',            ''],
-        ['systemd-reboot.service',              ''],
-        ['systemd-soft-reboot.service',         ''],
-        ['systemd-rfkill.socket',               'ENABLE_RFKILL'],
-        ['systemd-sysext.service',              'ENABLE_SYSEXT'],
-        ['systemd-confext.service',             'ENABLE_SYSEXT'],
-        ['systemd-sysupdate.timer',             'ENABLE_SYSUPDATE'],
-        ['systemd-sysupdate-reboot.timer',      'ENABLE_SYSUPDATE'],
-        ['systemd-sysusers.service',            'ENABLE_SYSUSERS',
-         'sysinit.target.wants/'],
-        ['systemd-tmpfiles-clean.service',      'ENABLE_TMPFILES'],
-        ['systemd-tmpfiles-clean.timer',        'ENABLE_TMPFILES',
-         'timers.target.wants/'],
-        ['systemd-tmpfiles-setup-dev.service',  'ENABLE_TMPFILES',
-         'sysinit.target.wants/'],
-        ['systemd-tmpfiles-setup.service',      'ENABLE_TMPFILES',
-         'sysinit.target.wants/'],
-        ['systemd-udevd-control.socket',        '',
-         'sockets.target.wants/'],
-        ['systemd-udev-settle.service',         ''],
-        ['systemd-udev-trigger.service',        '',
-         'sysinit.target.wants/'],
-        ['systemd-udevd-kernel.socket',         '',
-         'sockets.target.wants/'],
-        ['systemd-userdbd.socket',              'ENABLE_USERDB'],
-        ['time-set.target',                     ''],
-        ['time-sync.target',                    ''],
-        ['timers.target',                       ''],
-        ['tmp.mount',                           '',
-         'local-fs.target.wants/'],
-        ['umount.target',                       ''],
-        ['usb-gadget.target',                   ''],
-        ['user.slice',                          ''],
-        ['var-lib-machines.mount',              'ENABLE_MACHINED',
-         'remote-fs.target.wants/ machines.target.wants/'],
-        ['systemd-oomd.socket',                 'ENABLE_OOMD'],
+        { 'file' : 'basic.target' },
```

Comment:
Isn't this a bit redundant? Wouldn't it be simpler to change the list to a dictionary with unit name as a key? I.e.,
```python
units = {
        'basic.target': {},
        ... 
        'cryptsetup-pre.target': { 'conditions': ['HAVE_LIBCRYPTSETUP'] },
        ...
}

Reply:
Ah, it follows the same pattern as `executables`.